src/types/config.ts

```diff
@@ -0,0 +1,7 @@
+import { InferType } from 'yup';
+import { tagSchema, functionSchema, functionConfigSchema, configSchema } from '../config';
+
+export type FunctionConfig = InferType<typeof functionSchema>;
+export type Config = InferType<typeof configSchema>;
+export type Functions = InferType<typeof functionConfigSchema>;
+export type Tags = InferType<typeof tagSchema>;
```

Comment:
Just curious, what's the use of these types?

Reply:
We need these types for compile time, schemas are not types but variables, so we need a way to extract types from them and use them in our code.